cid/common.py

```diff
@@ -943,79 +943,93 @@ def create_or_update_dataset(self, dataset_definition: dict, dataset_id: str=Non
         cur_required = dataset_definition.get('dependsOn', dict()).get('cur')
         athena_datasource = None
 
-        if not len(self.qs.athena_datasources):
+        if get_parameters().get('quicksight-datasource-id'):
+            # We have explicit choice of datasource
+            datasource_id = get_parameters().get('quicksight-datasource-id')
+
+            try:
+                athena_datasource = self.qs.describe_data_source(datasource_id)
+            except self.qs.client.exceptions.AccessDeniedException:
+                athena_datasource = Datasource(raw={
```

Comment:
Why do we use the ID anyway even we don't have access to it?

Reply:
Here user provides explicitly the id. We do not need access to use it. User can just have a describe permission denied